Solink.AddIn.Helpers/RestartableAddInGenerator.cs

```diff
@@ -37,6 +37,7 @@ public RestartableAddInGenerator(string namespaceName, FileInfo sourceAssembly,
 
         public void Generate()
         {
+            AppDomain.CurrentDomain.ReflectionOnlyAssemblyResolve += (sender, args) => Assembly.ReflectionOnlyLoad(args.Name);

```

Comment:
There aren't too many ways of hooking the `ReflectionOnlyAssembleResolve` event, so that looks OK.

It would be even better if you could write a test that failed without this new line, perhaps by creating an `IThingWithDependency` interface that uses a foreign type (i.e. something not from `mscorlib` but from a dependent assembly) in one of its members, such as a method that accepts an instance of `AssertFailedException` (which comes from the `Microsoft.VisualStudio.QualityTools.UnitTestFramework` assembly).  I'm pretty sure just _adding_ that interface to the test project will cause the `Generate()` test to fail with a `FileLoadException`.